src/main/scala/org/bitcoins/core/protocol/script/ScriptPubKey.scala

```diff
@@ -26,14 +29,21 @@ sealed trait ScriptPubKey extends NetworkElement with BitcoinSLogger {
    */
   lazy val asm : Seq[ScriptToken] = ScriptParser.fromBytes(bytes)
 
+  def p2shAddress(network : NetworkParameters) : P2SHAddress = {
```

Comment:
Agree, I don't think it's placed appropriately, or if it's even necessary at all.